lib/authie/session.rb

```diff
@@ -1,338 +1,264 @@
-      active.where('expires_at IS NULL AND last_activity_at < ?',
-                   Authie.config.session_inactivity_timeout.ago).each(&:invalidate!)
-      # Invalidate persistent sessions that have expired
-      active.where('expires_at IS NOT NULL AND expires_at < ?', Time.now).each(&:invalidate!)
-      Authie.config.events.dispatch(:after_cleanup)
-      true
-    end
-
-    # Return a hash of a given token
-    def self.hash_token(token)
-      Digest::SHA256.hexdigest(token)
-    end
-
-    # Convert all existing active sessions to store their tokens in the database
-    def self.convert_tokens_to_hashes
-      active.where(token_hash: nil).where('token is not null').each do |s|
-        hash = hash_token(s.token)
-        where(id: s.id).update_all(token_hash: hash, token: nil)
-      end
-    end
-
-    private
-
-    # Return all cookies on the associated controller
-    def cookies
-      controller.send(:cookies)
-    end
+      delegate :hash_token, to: SessionModel
+    end
+
+    # Backwards compatibility with Authie < 4.0. These methods were all available on sessions
+    # in previous versions of Authie. They have been maintained for backwards-compatibility but
+    # will be removed entirely in Authie 5.0.
+    alias check_security! validate
+    alias persist! persist
+    alias invalidate! invalidate
+    alias touch! touch
+    alias set_cookie! set_cookie
+    alias see_password! see_password
+    alias mark_as_two_factored! mark_as_two_factored
+
+    # Delegate key methods back to the underlying session model. Previous behaviour in Authie
+    # exposed all methods on the session model. It is useful that these methods can be accessed
+    # easily from this session proxy model so these are maintained as delegated methods.
+    delegate :active?, to: :session
+    delegate :browser_id, to: :session
+    delegate :expired?, to: :session
+    delegate :first_session_for_browser?, to: :session
+    delegate :first_session_for_ip?, to: :session
+    delegate :get, to: :session
+    delegate :inactive?, to: :session
+    delegate :invalidate_others!, to: :session
+    delegate :last_activity_at, to: :session
+    delegate :last_activity_ip, to: :session
+    delegate :last_activity_path, to: :session
+    delegate :login_at, to: :session
+    delegate :login_ip, to: :session
+    delegate :password_seen_at, to: :session
+    delegate :persisted?, to: :session
+    delegate :persistent?, to: :session
+    delegate :recently_seen_password?, to: :session
+    delegate :requests, to: :session
+    delegate :set, to: :session
+    delegate :temporary_token, to: :session
+    delegate :token_hash, to: :session
+    delegate :two_factored_at, to: :session
+    delegate :two_factored_ip, to: :session
+    delegate :two_factored?, to: :session
+    delegate :update, to: :session
+    delegate :update!, to: :session
+    delegate :user_agent, to: :session
+    delegate :user, to: :session
```

Comment:
Correct me if I'm wrong, but I believe the `delegate` method here comes from ActiveSupport, meaning this class will fail to parse unless ActiveSupport/Rails has already been loaded.

I'd suggest adding `activesupport` as a runtime gem dependency, and then adding this require call to the top of the file to ensure this works in isolation:

```ruby
require 'active_support/core_ext/module/delegation'
```